usl_pipeline/usl_lib/usl_lib/shared/suppliers.py

```diff
@@ -0,0 +1,49 @@
+import abc
+import os
+
+
+class ChunkFilePathGenerator(abc.ABC):
+    """Abstract layer that can be used to generate paths for chunk files."""
+
+    @abc.abstractmethod
+    def generate(self, y_chunk_index: int, x_chunk_index: int) -> str:
+        """Generates path for a chunk file for given positional indices.
+
+        Args:
+            y_chunk_index: Index along the Y-axis for a chunk position.
+            x_chunk_index: Index along the X-axis for a chunk position.
+
+        Returns:
+            A path to a chunk file.
+        """
+        pass
+
+
+class PatternBasedChunkFilePathGenerator(ChunkFilePathGenerator):
+    """An implementation of a chunk file path generator based on file name pattern."""
+
+    def __init__(self, dir_path: str, file_name_pattern: str):
+        """Constructor.
+
+        Args:
+            dir_path: A directory where every chunk file path will point.
+            file_name_pattern: A pattern for chunk file name that will be used in
```

Comment:
Does this need to be variable? When would we want to vary the pattern?

Reply:
Removed the whole generator thingy as you suggested.